openedx/core/djangoapps/enrollments/views.py

```diff
@@ -987,3 +995,113 @@ def get_queryset(self):
         if usernames:
             queryset = queryset.filter(user__username__in=usernames)
         return queryset
+
+
+class EnrollmentAllowedView(APIView):
+    """
+    A view that allows the retrieval and creation of enrollment allowed for a given user email and course id.
+    """
+    authentication_classes = (
+        JwtAuthentication,
+    )
+    permission_classes = (IsStaffOrOwner,)
```

Comment:
I'm not 100% clear on who the intended audience of this endpoint is, but assuming it is for server-to-server calls, I'm worried that this permission class does not do what is intended.

AFAIK, `IsStaffOrOwner` [[GitHub](https://github.com/eduNEXT/edx-platform/blob/03299af16848496ff2a2228d59ef92dc8e173e56/openedx/core/lib/api/permissions.py#L119)] allows a request if a parameter called `user` or `username` is sent in that matches the username of the user making the request.

Did you mean to use `permissions.IsAdminUser`?

Reply:
I was going to suggest we use `ApiKeyHeaderPermissionIsAuthenticated` as most of the other views in this context, but I went back to `CourseEnrollmentsApiListView` which uses `IsAdminUser` and it is meant for administrative use. Given this is the same server-to-server admin use that we are proposing for this view I agree with Phillip's suggestion.

Reply:
I'll use IsAdminUser, thanks.